code/controllers/subsystem/stat.dm

```diff
@@ -0,0 +1,75 @@
+#define FLAT_ICON_CACHE_MAX_SIZE 250
+
+SUBSYSTEM_DEF(stat)
+	name = "Stat"
+	wait = 2 SECONDS
+	priority = FIRE_PRIORITY_STAT
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME	//RUNLEVEL_INIT doesn't work, so the stat panel will not auto update during this time (But that is good since we don't want to waste processing time during that phase).
+	init_order = INIT_ORDER_STAT
+	flags = SS_NO_INIT | SS_BACKGROUND
+
+	var/list/flat_icon_cache = list()	//Assoc list, datum = flat icon
+	var/list/currentrun = list()
+
+/datum/controller/subsystem/stat/fire(resumed = 0)
+	if (!resumed)
+		src.currentrun = GLOB.clients.Copy()
+
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+
+	while(currentrun.len)
+		var/client/C = currentrun[currentrun.len]
+		currentrun.len--
+
+		if (C)
+			var/mob/M = C.mob
+			if(M)
+				//Auto-update, not forced
+				M.UpdateMobStat(FALSE)
+
+		if (MC_TICK_CHECK)
+			return
+
+//Note: Doesn't account for decals on items.
+//Whoever examins an item with a decal first, everyone else will see that items decals.
+//Significantly reduces server lag though, like MASSIVELY!
+/datum/controller/subsystem/stat/proc/get_flat_icon(atom/A)
+	var/what_to_search = "[A.type][A.icon_state[1]]"
```

Comment:
```suggestion
	var/what_to_search = "[A.type][A.icon_state ? A.icon_state[1] : "*"]"
```

It turns out, tables, full tile windows and shits do have an empty string as `icon_state`. The whole proc runtimes and return when it tries to access the first letter of an empty string.

Reply:
Yikes